src/matching/Generic_vs_generic.ml

```diff
@@ -2210,15 +2196,7 @@ and m_attributes a b = m_list_in_any_order ~less_is_ok:true m_attribute a b
 (* experimental! *)
 and m_stmts_deep ~inside ~less_is_ok (xsa : G.stmt list) (xsb : G.stmt list) tin
     =
-  (* shares the cache with m_list__m_stmt *)
-  match (tin.cache, xsa, xsb) with
-  | Some cache, a :: _, _ :: _ when a.s_use_cache ->
-      let tin = { tin with mv = Env.update_min_env tin.mv a } in
-      Caching.Cache.match_stmt_list ~access:cache_access ~cache
-        ~function_id:CK.Match_deep ~list_kind:CK.Original ~less_is_ok
-        ~compute:(m_stmts_deep_uncached ~inside ~less_is_ok)
-        ~pattern:xsa ~target:xsb tin
-  | _ -> m_stmts_deep_uncached ~inside ~less_is_ok xsa xsb tin
+  m_stmts_deep_uncached ~inside ~less_is_ok xsa xsb tin
```

Comment:
let's rename later those m_xxx_uncached and remove the uncached suffix (in another PR).

Reply:
Since I have to fix the test failure I'll do this now too.